### DISCUSSION TITLE
Erro ao acessar Req.Session no Lazarus

### DISCUSSION THREAD

**Discussion by VagnerBecker** [HashLoad/horse#269] [General]
Bom dia,

Estou com dificuldades para extrair as informações da chave JWT no Lazarus. Pelo que entendi na documentação os dados da session estão em formato Json então estou fazendo um parse da informação.... Segue um exemplo de captura dos dados:

    jo := TJsonObject.Create;
    jo.Parse(Req.Session);
    writeln(jo.Stringify);

Retorna o seguinte erro:

     router.authenticate.pas(56,18) Error: Wrong number of parameters specified for call to "Session"
     Horse.Request.pas(214,24) Error: Found declaration: Session(const TObject):THorseRequest;

Estou usando a última versão do horse.

Abraço,

Vagner Felipe Becker

**Comment by VagnerBecker** [comment]
Respondendo minha própria pergunta. Não sei se é a melhor forma, porém achei uma maneira de acessar a chave JWT e extrair o conteúdo da claim sub.

Não esqueçam de adicionar a unit "fpjwt"

```
function GetClaimSUB(Req: THorseRequest): string;
var
  JWT: TJWT;
  ClaimSUB, Token: string;
begin
  Token := Req.Headers['authorization'];
  Token := Trim(Token.Replace('bearer', '', [rfIgnoreCase]));
  try
    JWT := TJWT.Create;
    JWT.AsString := Token;
    ClaimSUB := JWT.Claims.sub;
  finally
    FreeAndNil(JWT);
  end;
  Result := ClaimSUB;
end; 
```

**Reply by rickdroio** [reply]
sensacional! estava tentando pelo session de todas as maneiras possiveis e não funcionou.
JWT pra Lazarus está bem limitado mesmo, Delphi está bem na frente.

@VagnerBecker vc conseguiu colocar custom claims?
